nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/forms/FormData.java

```diff
@@ -221,16 +226,25 @@ public Blob getFirstBlob() {
 
     protected Blob getBlob(FileItem item) {
         try {
-            InputStream in;
+            Blob blob;
             if (item.isInMemory()) {
-                in = new ByteArrayInputStream(item.get());
+                blob = Blobs.createBlob(item.get());
             } else {
-                in = item.getInputStream();
+                File file;
+                if (item instanceof DiskFileItem //
+                        && (file = ((DiskFileItem) item).getStoreLocation()) != null) {
```

Comment:
I am not keen on assignment inside an if clause, but clearly you are, and it saves an additional "if".

Reply:
This pattern avoids a rather more complex code with duplication, given that there's an `else` statement.